File: src/katana_mcp/widgets/product_list.py
from __future__ import annotations

from pathlib import Path

from .registry import WidgetSpec

WIDGET_VERSION = "v3"
PRODUCT_LIST_URI = f"ui://widget/product-list-{WIDGET_VERSION}.html"


def product_list_widget(web_dist: Path) -> WidgetSpec:
    return WidgetSpec(
        uri=PRODUCT_LIST_URI,
        name="PRoduct List Widget",
        description="React widget rendering the interactive product list.",
        html_path=web_dist / "index.html",
        meta={
            "openai/widgetDescription": "Interactive product board.",
            "openai/widgetPrefersBorder": True,
        },
    )

File: apps/katana-mcp/resources/widget/productlist.go
package widget

import (
	"context"

	"github.com/mark3labs/mcp-go/mcp"
)

func ProductListWidget(productListHTML string) (mcp.Resource, MCPResourceHandler) {

	if productListHTML == "" {
		productListHTML = `<!doctype html><meta charset='utf-8'><div style='font-family:sans-serif;padding:1rem'>Widget bundle missing. Build it first.</div>`
	}

	productListMeta := map[string]any{
		"openai/widgetDescription":   "Interactive Katana product list.",
		"openai/widgetPrefersBorder": true,
		"openai/widgetAccessible":    true,
	}

	productListResource := mcp.NewResource(
		PRODUCT_LIST_URI,
		"Product List Widget",
		mcp.WithResourceDescription("Displays a searchable product list of Katana products"),
		mcp.WithMIMEType("text/html+skybridge"),
	)
	productListResource.Meta = mcp.NewMetaFromMap(productListMeta)

	productListResourceHandler := func(ctx context.Context, request mcp.ReadResourceRequest) ([]mcp.ResourceContents, error) {
		return []mcp.ResourceContents{
			mcp.TextResourceContents{
				URI:      PRODUCT_LIST_URI,
				MIMEType: "text/html",
				Text:     productListHTML,
				Meta:     productListMeta,
			},
		}, nil
	}

	return productListResource, productListResourceHandler

}
